src/plugins/data/public/ui/query_editor/query_editor_extensions/query_editor_extensions.tsx

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo } from 'react';
+import {
+  QueryEditorExtension,
+  QueryEditorExtensionConfig,
+  QueryEditorExtensionDependencies,
+} from './query_editor_extension';
+
+interface QueryEditorExtensionsProps extends QueryEditorExtensionDependencies {
+  configMap?: Record<string, QueryEditorExtensionConfig>;
```

Comment:
q: is there a reason this has to be a map vs an array?

Reply:
consistency with existing code, see https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7034#discussion_r1641543079